src/hardware/calibration.py

```diff
@@ -170,34 +182,41 @@ def interpolate_2d(self, x, y, x1, x2, y1, y2, z11, z21, z12, z22):
         return self.interpolate_1d(y, y1, y2, z_y1, z_y2)
 
 
+def freq_to_compare(f):
+    """Allow a frequency of type [float] to be compared with =="""
+    f = int(round(f))
+    return f
+
+
 def load_from_json(fname):
-    with open(fname) as f:
-        calibration = json.load(f)
+    with open(fname) as file:
+        calibration = json.load(file)
+        file.close()
 
+    # Check that the required fields are in the dict
     assert "calibration_datetime" in calibration
     assert "calibration_frequency_divisions" in calibration
-    assert "calibration_points" in calibration
+    assert "calibration_data" in calibration
     assert "clock_rate_lookup_by_sample_rate" in calibration
 
-    last_frequency = 0
-    last_gains = []
-    gains = []
+    # Load all the calibration data
     calibration_data = {}
-    for calibration_point in calibration["calibration_points"]:
-        frequency = calibration_point["freq_sigan"]
-        gain = calibration_point["gain_sigan"]
-        if frequency == last_frequency:
-            gains.append(gain)
-        else:
-            calibration_data[frequency] = {}
-            # gains should be equal for all calibration points
-            if last_gains and len(calibration_data) > 2:
-                assert gains == last_gains
-            last_gains = gains
-            gains = [gain]
-        last_frequency = frequency
-        calibration_data[frequency][gain] = calibration_point
-
+    for sample_rate_row in calibration["calibration_data"]["sample_rates"]:
```

Comment:
I think the current format still has the data in a list format, so we wouldn't be able to use the gain, frequency, and sample rate as dictionary keys. To reference the data, it would be:
```calibration['calibration_data']['sample_rates'][sr_i]['calibration_data']['frequencies'][f_i]['calibration_data']['gains'][g_i]```

One thing we may think about to resolve the JSON keys needing to be strings is to convert the gains, frequencies, and sample rates to strings, then converting them back to numbers when we load it back in. The structure would look like:

```{
...
'calibration_data': {
  '10000000.0': {
    '40.0': {
      '700000000.0': {
        'gain_sigan': 43.2934746,
        ...
      },
    },
  },
}```

This would only require converting the keys to numbers when we load it.